controller/array_action/array_mediator_ds8k.py

```diff
@@ -581,11 +576,9 @@ def delete_snapshot(self, snapshot_id):
         if not api_volume.flashcopy:
             logger.error(
                 "FlashCopy relationship not found for target volume: {}".format(api_volume.name))
-            raise array_errors.ExpectedSnapshotButFoundVolumeError(api_volume.name,
-                                                                   self.service_address)
-        self._assert_snapshot_is_not_in_use(snapshot_id, api_volume.flashcopy)
-        for flashcopy in api_volume.flashcopy:
-            self._delete_flashcopy(flashcopy.id)
+            raise array_errors.VolumeNameBelongsToSnapshotError(api_volume.name,
```

Comment:
care to explain?

Reply:
dont know, changed for NotFound in case of snapshot is not recognized as one.